src/dawn/Optimizer/PassInlining.cpp

```diff
@@ -285,7 +285,7 @@ class Inliner : public ASTVisitor {
         std::iter_swap(newStmtAccessesPairs_.begin() + stmtIdxOfFunc,
                        std::prev(newStmtAccessesPairs_.end()));
 
-        replaceOldExprWithNewExprInStmt(
+        iir::replaceOldExprWithNewExprInStmt(
```

Comment:
this should not be specific but just take both of them (since it takes base ptr's. Otherwise we're duplicating code

Reply:
ptr's to iir::Stmt and ptr's to sir::Stmt cannot be cast one to each other, since they will be different specializations of the same class. But `replaceOldExprWithNewExprInStmt()` is going to be templated in the same way to avoid code duplication, and have its iir:: and sir:: specializations define on which Stmt specialization to operate on.